Silence the following warning

url.c: In function 'Curl_connect':
url.c:2013:13: warning: offset outside bounds of constant string
     hostname++;
             ^

--- lib/url.c
+++ lib/url.c
@@ -1893,6 +1893,7 @@
   CURLU *uh;
   CURLUcode uc;
   char *hostname;
+  char empty_str[] = "";
 
   up_free(data); /* cleanup previous leftovers first */
 
@@ -1999,7 +2000,7 @@
   hostname = data->state.up.hostname;
   if(!hostname)
     /* this is for file:// transfers, get a dummy made */
-    hostname = (char *)"";
+    hostname = empty_str;
 
   if(hostname[0] == '[') {
     /* This looks like an IPv6 address literal. See if there is an address
